support/cmake/fmt-config.cmake.in

```diff
@@ -5,3 +5,31 @@ if (NOT TARGET fmt::fmt)
 endif ()
 
 check_required_components(fmt)
+
+function(add_fmt_module NAME)
```

Comment:
I don't think we should be duplicating the logic to create the fmt module.

Reply:
The it will not work to use the install module:
```bash
[6/7] /usr/local/opt/llvm/bin/clang++ -DFMT_MODULE -isystem /usr/local/include -std=c++20 -std=c++23 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk -stdlib=libc++ -MD -MT CMakeFiles/module-test.dir/main.cc.o -MF CMakeFiles/module-test.dir/main.cc.o.d @CMakeFiles/module-test.dir/main.cc.o.modmap -o CMakeFiles/module-test.dir/main.cc.o -c /Users/clausklein/Workspace/cpp/fmt/test/find-package-test/main.cc
FAILED: CMakeFiles/module-test.dir/main.cc.o 
/usr/local/opt/llvm/bin/clang++ -DFMT_MODULE -isystem /usr/local/include -std=c++20 -std=c++23 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk -stdlib=libc++ -MD -MT CMakeFiles/module-test.dir/main.cc.o -MF CMakeFiles/module-test.dir/main.cc.o.d @CMakeFiles/module-test.dir/main.cc.o.modmap -o CMakeFiles/module-test.dir/main.cc.o -c /Users/clausklein/Workspace/cpp/fmt/test/find-package-test/main.cc
error: C++23 was disabled in PCH file but is currently enabled
error: module file CMakeFiles/fmt__fmt@synth_8c418311d24f.dir/ad882f9916e2.bmi cannot be loaded due to a configuration mismatch with the current compilation [-Wmodule-file-config-mismatch]
/Users/clausklein/Workspace/cpp/fmt/test/find-package-test/main.cc:3:2: error: expected unqualified-id
    3 | #else
      |  ^
3 errors generated.
ninja: build stopped: subcommand failed.
bash-5.2$ 
```

Reply:
For the installed case, `find_package` should take care of the necessary configuration. If the installed targets are somehow not usable directly we should fix that and not try recreating them here.

Reply:
This is not possible! You compile the `fmt` library with one compiler and c++ standard if you install it.
No other compiler or compiler options can be used for the c++ module!

Please try the example from  https://github.com/ClausKlein/cxx-modules/blob/master/cmake/readme.md

Reply:
Sure but this doesn't mean that we have to duplicate the build config in the test. It means that the test should check if the installed package is compatible, otherwise you are testing the wrong thing.

Reply:
> It means that the test should check if the installed package is compatible, otherwise you are testing the wrong thing.

Do you have an idea how to check this?
If compilation or linking fails, it does not have a compatible setting...; and than?